packages/mds-web-sockets/ws-server.ts

```diff
@@ -105,6 +105,6 @@ export const WebSocketServer = async <T extends readonly string[]>(entityTypes?:
   }
 
   supportedEntities.forEach(async e => {
-    await stream.NatsStreamConsumer(`${TENANT_ID}.event`, processor).initialize()
+    await stream.NatsStreamConsumer(`${TENANT_ID}.${e}`, processor).initialize()
   })
```

Comment:
This feels like it should probably be 
```
await Promise.all(
  supportedEntities.map(e => stream.NatsStreamConsumer(`${TENANT_ID}.${e}`, processor).initialize()
)
```

Reply:
When I try to switch over to a map call as suggested @drtyh2o I get the following error on it.  I'm not quite grokking.  @avatarneil I know you want to land this today perhaps you have thoughts?  Or I can land without.  I don't have time to dive in on it at the moment.

This expression is not callable.
  Each member of the union type '(<U>(callbackfn: (value: string, index: number, array: readonly string[]) => U, thisArg?: any) => U[]) | (<U>(callbackfn: (value: "event" | "telemetry", index: number, array: readonly ("event" | "telemetry")[]) => U, thisArg?: any) => U[])' has signatures, but none of those signatures are compatible with each other.ts(2349)

Reply:
It's interesting that forEach worked and map does not, perhaps forEach is a method that's more broadly supported

Reply:
Here's the type signature

async <T extends readonly string[]>(entityTypes?: T)

Reply:
This works...

` 
await Promise.all(
    (supportedEntities as readonly string[]).map(e =>
      stream.NatsStreamConsumer(`${TENANT_ID}.${e}`, processor).initialize()
    )
  )`